Link Host certificates if available in Control Container

sumukhballal opened the pull request:
<!--
Tips:
- Please read the CONTRIBUTING document to understand our process and our requests for PRs.
- Please file an issue before creating a PR so we can discuss the change and confirm it's not already being worked on.
-->

**Description of changes:**

This change links host certificates if it has been updated on the Bottlerocket host. It checks if the path /.bottlerocket/certs exists, if so links the certs to the ca-trust source.

- [x] Test variant aws-k8s-1.28 boot
- [x] Test variant aws-k8s-1.24 boot
- [x] Container specific tests


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

jpculp commented on start_control_ssm.sh:
```diff
@@ -6,6 +6,16 @@ declare -r PERSISTENT_STORAGE_BASE_DIR="/.bottlerocket/host-containers/current"
+declare -r HOST_CERTS="/.bottlerocket/certs"
+
+# Link host certs if present into container & run update-ca-trust
+if [ -e "${HOST_CERTS}" ]; then
```
```suggestion
if [[ -d "${HOST_CERTS}" ]]; then
```

jpculp commented on start_control_ssm.sh:
```diff
@@ -6,6 +6,16 @@ declare -r PERSISTENT_STORAGE_BASE_DIR="/.bottlerocket/host-containers/current"
 declare -r SSM_AGENT_LOCAL_STATE_DIR="/var/lib/amazon/ssm"
+declare -r HOST_CERTS="/.bottlerocket/certs"
+
+# Link host certs if present into container & run update-ca-trust
```
Can we move this block to after the function declarations, or make it a function itself, to remain consistent with the rest of the script?

sumukhballal committed:
certs: link host certs if present

In certain cases, if host certificates are present at the path '/.bottlerocket/certs', we would link them to the
ca-trust and update-ca-trust process to pickup the new certificates.

sumukhballal commented:
Addressed @jpculp 's comments.

jpculp commented on start_control_ssm.sh:
```diff
@@ -55,6 +67,8 @@ fetch_from_json() {
 # then check to see if the user-data file contains ssm at the top-level. If so,
 # attempt to manually register with SSM with a hybrid activation.
 
+link_host_certs
```
```suggestion
[[ -d "${HOST_CERTS}" ]] && link_host_certs
```
nit, but it seems more clear to have the conditional outside the function.

sumukhballal commented:
Addressed @jpculp 's comment.
